src/csg/csg.jl

```diff
@@ -116,7 +116,7 @@ end
 - [`@pcsgrammar`](@ref) uses a similar syntax to create probabilistic [`ContextSensitiveGrammar`](@ref)s.
 """
 macro csgrammar(ex)
-	return expr2csgrammar(ex)
+	return :(expr2csgrammar($(QuoteNode(ex))))
```

Comment:
Could you describe/link the issue which this is solving? In what cases do we not get an expression here?

Reply:
In all cases. `expr2csgrammar(ex)` returns a `ContextSensitiveGrammar` object. Example (HerbGrammar v0.3.0):

```julia
julia> res = @macroexpand @csgrammar begin
               A = 1
       end
1: A = 1


julia> typeof(res)
ContextSensitiveGrammar
```

With this change:

```julia
julia> res = @macroexpand @csgrammar begin
           A = 1
       end
:(HerbGrammar.expr2csgrammar($(QuoteNode(quote
    #= REPL[4]:2 =#
    A = 1
end))))

julia> typeof(res)
Expr

julia>
```

This makes it possible to set a breakpoint in the `expr2csgrammar` method and use a debugger. The debugger (at least in VSCode) doesn't break on any code that runs within the macro expansion step.

Generally, the pattern for macros is to modify the code and return the new `Expr`. See `@show`:

```julia
julia> x = "Hello, world!"
"Hello, world!"

julia> res = @macroexpand @show x
quote
    Base.println("x = ", Base.repr(begin
                #= show.jl:1181 =#
                local var"#3#value" = x
            end))
    var"#3#value"
end

julia> typeof(res)
Expr
```